### ISSUE TITLE
WSL2 update hangs up without any log info

### ISSUE THREAD

**Issue by EmperorTransisthor** [microsoft/WSL#10943]
### Windows Version

Microsoft Windows [Version 10.0.19045.3803]

### WSL Version

2.0.14.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.133.1-microsoft-standard-WSL2

### Distro Version

Ubuntu 22.04

### Repro Steps

> Run powershell (without admin priviledges)
> Run command $ wsl --update

### Expected Behavior

Display error: "Can't update Windows Subsystem Linux, Powershell needs elevated priviledges. Please run Powershell as admin."

### Actual Behavior

Powershell tells that wsl is now updating to version X
Nothing happens, and process is running in the background using 5% of CPU power.

**Comment by OneBlue** [comment]
@EmperorTransisthor: Are you seeing any error or is wsl --update hanging ? Admin access is indeed required to update WSL so what you're seeing is the expected behavior.

**Comment by EmperorTransisthor** [comment]
> Maybe I'm misunderstanding your issue completely, in this case I apologize, but the error message you're getting is quite clear?

The thing is, that I get no error message at all. Correct me, if I'm wrong, but in my opinion it is an issue, because it is misleading. Only after I killed operation with ^C, it got unstucked and told me there was an error and gave me logs, from where I got feedback that it got stuck due to access denies, thus missing elevated priviledges.

> Have you tried running wsl --update from an admin command prompt or powershell?

Yes, actually this issue on github is more of a potential defect, not a serious issue. I got my wsl successfully updated after running powershell with admin priviledges.

> Also, according to the 2nd reproduction step, it looks like you're running wsl from inside wsl? You write `Run command $ wsl --update`, does the `$` mean you're running the wsl command from a Linux shell?

Sorry for misunderstanding, if '$' prefix is not used for citing cmd/powershell commands. I've got used to citing scripts I use with '$' prefix. I run it in powershell without admin rights, and typed in `wsl --update`

@OneBlue no, I don't see any error at all. From my point of view, this is an issue due to it's lack of any feedback given to user, therefore misleading behaviour which can make user believe, that update is ongoing (which was in my case btw), however it is not. Solution would be to check if command is running with sufficient priviledges and if not, raise error "insufficient priviledges". What's your take on that?
